### PR TITLE
Adding prompt_toolkit

### PR DESCRIPTION
**Pull request by ocefpaf**
Hopefully this will fix `xonsh` build.

### CODE REVIEW

**Comment by ocefpaf** [comment]
If I understand the problem correctly `prompt_toolkit` uses `fcntl`, and`fcntl` is not available on Windows.

However, the PyPI page for `prompt_toolkit` says:
  `Runs on Linux, OS X, OpenBSD and Windows systems.`

I lost here. Any clues @scopatz?

File: recipes/prompt_toolkit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        - wcwidth
+
+test:
+    imports:
+        - prompt_toolkit
+        - prompt_toolkit.clipboard
+        - prompt_toolkit.contrib
+        - prompt_toolkit.contrib.completers
+        - prompt_toolkit.contrib.regular_languages
+        - prompt_toolkit.contrib.telnet
```
Comment (scopatz) [inline recipes/prompt_toolkit/meta.yaml]: I think that you need to remove this line or add a `[not win]` selector.  This subpackage probably does not apply to windows.

**Comment by scopatz** [comment]
> I lost here. Any clues @scopatz?

Comment in-line.  Prompt-toolkit definitely works on windows. You are probably just hitting an invalid test case.

File: recipes/prompt_toolkit/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline recipes/prompt_toolkit/meta.yaml]: Maybe we should send a patch upstream to return a more informative message here.  Fixing this now...

**Comment by ocefpaf** [comment]
And we have green!

@scopatz can you review and merge this one?

**Comment by scopatz** [comment]
Woohoo thanks!

**Comment by pelson** [comment]
P.S. It might be worth pinging the devs so that they know their package has been community built. :+1:

**Comment by scopatz** [comment]
I think that @jonathanslenders is the relevant person in this instance, but I'll post to their issue tracker too.

**Pull request merged**